Ghidra/Features/Decompiler/src/decompile/cpp/prettyprint.cc

```diff
@@ -291,9 +291,16 @@ void EmitXml::tagLabel(const char *ptr,syntax_highlight hl,
 void EmitXml::print(const char *str,syntax_highlight hl)
 
 {
-  *s << "<syntax " << highlight[(int4)hl] << '>';
-  xml_escape(*s,str);
-  *s << "</syntax>";
+  if ('\0' == *str && no_color == hl)
+    return;
+  *s << "<syntax" << highlight[(int4)hl];
+  if ('\0' == *str)
+    *s << "/>";
+  else {
+    *s << '>';
+    xml_escape(*s,str);
+    *s << "</syntax>";
+  }
```

Comment:
See https://github.com/NationalSecurityAgency/ghidra/pull/1702#issuecomment-607577363. I'm not 100% sure but the space may be needed if there is `color` in the tag.

Reply:
You are right! The missing space between the "syntax" and "color" in the **after** image is my mistake when I reformatted the response by hand. In the real response everything is OK.